app/gui/view/component-browser/searcher-list-panel/src/navigator.rs

```diff
@@ -0,0 +1,255 @@
+//! The section navigator bar. This is a narrow bar on the left of the Searcher List Panel that
+//! contains two sets of navigation buttons.
+//!
+//! See [`Navigator`] struct.
+
+use ensogl_core::display::shape::*;
+use ensogl_core::prelude::*;
+
+use crate::component_group::icon;
+use crate::Style;
+
+use enso_frp as frp;
+use ensogl_core::application::Application;
+use ensogl_core::data::color;
+use ensogl_core::display;
+use ensogl_core::display::scene::Layer;
+use ensogl_core::display::style;
+use ensogl_core::display::style::Path;
+use ensogl_hardcoded_theme::application::component_browser::searcher as searcher_theme;
+use ensogl_list_view as list_view;
+use ensogl_list_view::entry::AnyModelProvider;
+use ensogl_shadow as shadow;
+use searcher_theme::list_panel as list_panel_theme;
+
+
+
+// =============
+// === Style ===
+// =============
+
+/// A navigator-specific style parameters.
+#[derive(Clone, Debug, Default)]
+pub struct NavigatorStyle {
+    pub size:            Vector2,
+    pub top_padding:     f32,
+    pub bottom_padding:  f32,
+    pub list_view_width: f32,
+}
+
+
+
+// ==============
+// === Shadow ===
+// ==============
+
+/// A shadow between the navigator bar and the main part of the Searcher List Panel.
+///
+/// We should have this shape embedded into the background shape, but we use a separate object
+/// because of https://www.pivotaltracker.com/story/show/182593513.
+pub mod navigator_shadow {
+    use super::*;
+
+    ensogl_core::define_shape_system! {
+        above = [crate::background];
+        (style:Style) {
+            let theme_path: style::Path = list_panel_theme::HERE.into();
+            let content_height = style.get_number(theme_path.sub("content_height"));
+            let menu_height = style.get_number(theme_path.sub("menu_height"));
+            let navigator_width = style.get_number(theme_path.sub("navigator").sub("width"));
+            let height = content_height + menu_height;
+            let width = navigator_width;
+            let base_shape = Rect((width.px(), height.px() * 2.0)).translate_x(width.px());
+            shadow::from_shape(base_shape.into(), style)
+        }
+    }
+}
+
+
+
+// ============
+// === Icon ===
+// ============
```

Comment:
The Icon model for list view deserves being more public - perhaps in a submodule of icon module.